src/transformers/modeling_utils.py

```diff
@@ -457,7 +458,11 @@ def load_state_dict(checkpoint_file: Union[str, os.PathLike]):
             )
         return safe_load_file(checkpoint_file)
     try:
-        return torch.load(checkpoint_file, map_location="cpu")
+        if is_deepspeed_zero3_enabled() and dist.is_initialized() and dist.get_rank() > 0:
```

Comment:
Shouldn't this be the `local_rank` here? We need to load the state dict once per machine or just once is okay?

Reply:
that's a good question. For ZeRO-DP just machine:0 rank:0 is enough, it then shards it to all gpus across multiple machines

But I'm yet to try ZeRO++ - @tjruwase, we haven't discussed this case. Won't loading on just machine:0 rank:0 become an issue if there are multiple DP levels in ZeRO++, so that it'd require loading of local_rank=0 on each node? and then the proposed code should be adapted to load on rank 0 of each node?

Reply:
I think just once should be okay, since the subsequent partitioning [here](https://github.com/huggingface/transformers/blob/9ef5256dfb28f5419649af8ea94e82573a6b2e77/src/transformers/modeling_utils.py#L539) is global.

Reply:
super, thank you for validating, Tunji!

Reply:
So all good for me, just reverting the `dist` rename and we can merge this :-)

Reply:
@stas00, I think this should also work for ZeRO++ because the multiple DP replicas are activated only during the backward pass to reduce allgather overheads.

Reply:
That's great to know, I appreciate you taking time to address this, Tunji!